src/test/java/com/vinylteam/vinyl/web/servlets/ShopServletTest.java

```diff
@@ -0,0 +1,123 @@
+    void doGetWithNoSessionTest() throws IOException {
+        //prepare
+        when(mockedShopService.findAll()).thenReturn(testListOfShops);
+        when(mockedRequest.getSession(false)).thenReturn(null);
+        when(mockedResponse.getWriter()).thenReturn(mockedPrintWriter);
+        //when
+        shopServlet.doGet(mockedRequest, mockedResponse);
+        //then
+        verify(mockedShopService).findAll();
+        inOrderRequest.verify(mockedRequest).getSession(false);
+        verify(mockedHttpSession, times(0)).getAttribute("user");
+        verify(mockedUser, times(0)).getRole();
+        inOrderResponse.verify(mockedResponse).setContentType("text/html;charset=utf-8");
+        inOrderResponse.verify(mockedResponse).setStatus(HttpServletResponse.SC_OK);
+        inOrderResponse.verify(mockedResponse).getWriter();
+    }
+
+    @Test
+    @DisplayName("Checks if all right methods are called & user is authed in ShopServlet`s get() method")
+    void doGetWithAuthedUserTest() throws IOException {
+        //prepare
+        when(mockedShopService.findAll()).thenReturn(testListOfShops);
+        when(mockedRequest.getSession(false)).thenReturn(mockedHttpSession);
+        when(mockedHttpSession.getAttribute("user")).thenReturn(mockedUser);
+        when(mockedUser.getRole()).thenReturn(Role.USER);
+        when(mockedResponse.getWriter()).thenReturn(mockedPrintWriter);
+        //when
+        shopServlet.doGet(mockedRequest, mockedResponse);
+        //then
+        verify(mockedShopService).findAll();
+        inOrderRequest.verify(mockedRequest).getSession(false);
+        verify(mockedHttpSession).getAttribute("user");
+        assertEquals(mockedUser, mockedHttpSession.getAttribute("user"));
+        verify(mockedUser).getRole();
+        assertEquals(Role.USER, mockedUser.getRole());
+        inOrderResponse.verify(mockedResponse).setContentType("text/html;charset=utf-8");
+        inOrderResponse.verify(mockedResponse).setStatus(HttpServletResponse.SC_OK);
+        inOrderResponse.verify(mockedResponse).getWriter();
+    }
+
+    @Test
+    @DisplayName("Checks if all right methods are called & user is not authed in ShopServlet`s get() method")
+    void doGetWithNotAuthedUserTest() throws IOException {
+        //prepare
+        when(mockedShopService.findAll()).thenReturn(testListOfShops);
+        when(mockedRequest.getSession(false)).thenReturn(mockedHttpSession);
+        when(mockedHttpSession.getAttribute("user")).thenReturn(null);
+        when(mockedResponse.getWriter()).thenReturn(mockedPrintWriter);
+        //when
+        shopServlet.doGet(mockedRequest, mockedResponse);
+        //then
+        verify(mockedShopService).findAll();
+        inOrderRequest.verify(mockedRequest).getSession(false);
+        verify(mockedHttpSession).getAttribute("user");
+        assertNull(mockedHttpSession.getAttribute("user"));
+        verify(mockedUser, times(0)).getRole();
```

Comment:
you can use never() in place of times(0) here.

Reply:
thanks!